--- a/src/obs-utils.c
+++ b/src/obs-utils.c
@@ -108,7 +108,7 @@
 		line_i++;
 		if (strncmp(line, "#include", 8) == 0) {
 			// Open the included file, place contents here.
-			char *pos = strrchr(file_name, '/');
+			const char *pos = strrchr(file_name, '/');
 			const size_t length = pos - file_name + 1;
 			struct dstr include_path = {0};
 			dstr_ncopy(&include_path, file_name, length);
--- a/src/obs-noise-source.c
+++ b/src/obs-noise-source.c
@@ -1577,7 +1577,7 @@
 	struct dstr shader_file;
 	dstr_init(&shader_file);
 	{
-		char *pos = strrchr(file_name, '/');
+		const char *pos = strrchr(file_name, '/');
 		const size_t length = pos - file_name + 1;
 		struct dstr include_path = {0};
 		dstr_ncopy(&include_path, file_name, length);
@@ -1595,7 +1595,7 @@
 		line_i++;
 		if (strncmp(line, "#include", 8) == 0) {
 				// Open the included file, place contents here.
-				char *pos = strrchr(file_name, '/');
+				const char *pos = strrchr(file_name, '/');
 				const size_t length = pos - file_name + 1;
 				struct dstr include_path = {0};
 				dstr_ncopy(&include_path, file_name, length);
